src/rpc/mempool.cpp

```diff
@@ -376,6 +376,90 @@ RPCHelpMan getmempoolentry()
+                                    {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                },
+                            },
+                        },
+                    },
+                },
+                RPCResult{
+                    RPCResult::Type::ARR, "", "",
+                    {
+                        {RPCResult::Type::OBJ, "", "",
+                        {
+                            {RPCResult::Type::STR_HEX, "txid", "the transaction id of the spent output"},
+                            {RPCResult::Type::NUM, "vout", "the vout value of the spent output"},
+                            {RPCResult::Type::STR_HEX, "spendingtxid", "the transaction id of the mempool transaction spending this output (omitted if unspent)"},
+                        }},
+                    }
+                },
+                RPCExamples{
+                    HelpExampleCli("getmempooltxspendingprevout", "\"[{\\\"txid\\\":\\\"a08e6907dbbd3d809776dbfc5d82e371b764ed838b5655e72f463568df1aadf0\\\",\\\"vout\\\":3}]\"")
+                    + HelpExampleRpc("getmempooltxspendingprevout", "\"[{\\\"txid\\\":\\\"a08e6907dbbd3d809776dbfc5d82e371b764ed838b5655e72f463568df1aadf0\\\",\\\"vout\\\":3}]\"")
+                },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    RPCTypeCheckArgument(request.params[0], UniValue::VARR);
+    const UniValue& output_params = request.params[0];
+    if (output_params.empty()) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, outputs are missing");
+    }
+
+    std::vector<COutPoint> outputs;
+    outputs.reserve(output_params.size());
+
+    for (unsigned int idx = 0; idx < output_params.size(); idx++) {
+        const UniValue& o = output_params[idx].get_obj();
+
+        RPCTypeCheckObj(o,
+            {
+                {"txid", UniValueType(UniValue::VSTR)},
+                {"vout", UniValueType(UniValue::VNUM)},
+            });
+
+        const uint256 txid(ParseHashO(o, "txid"));
+        const int nOutput = find_value(o, "vout").get_int();
+        if (nOutput < 0) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, vout cannot be negative");
+        }
+
+        const COutPoint output(txid, nOutput);
+        outputs.push_back(output);
+    }
+
+    const CTxMemPool& mempool = EnsureAnyMemPool(request.context);
+    LOCK(mempool.cs);
+
+    UniValue result{UniValue::VARR};
+
+    for (const COutPoint& output : outputs) {
+        UniValue o(UniValue::VOBJ);
+        o.pushKV("txid", output.hash.ToString());
+        o.pushKV("vout", (uint64_t)output.n);
+
+        const CTransaction* spendingTx = mempool.GetConflictTx(output);
+        if (spendingTx != nullptr) {
+            o.pushKV("spendingtxid", spendingTx->GetHash().ToString());
+        }
+
+        result.push_back(o);
+    }
```

Comment:
```suggestion
    for (const COutPoint& output : outputs) {
        result.emplace_back(UniValue::VOBJ);
        UniValue o = result.back();
        o.pushKV("txid", output.hash.ToString());
        o.pushKV("vout", (uint64_t)output.n);

        const CTransaction* spendingTx = mempool.GetConflictTx(output);
        if (spendingTx != nullptr) {
            o.pushKV("spendingtxid", spendingTx->GetHash().ToString());
        }
    }
```

In this way we avoid the move too, and for a transaction this is important IMHO

Reply:
Thanks for the review! This doesn't seem to compile though...and I'm not sure it's needed at all, we're not dealing with a whole transaction here, just a `txid`, so a copy is really not something we care about IMO.

Reply:
>Thanks for the review! This doesn't seem to compile though

I will try to compile it locally maybe something wrong with `UniValue::VOBJ` that my mind compiler is not able to catch right now!

>just a txid, so a copy is really not something we care about IMO

Right, I wrote a bad motivation, I mean here there `COutPoint` can be a lot, and this loop with the `emplace_back` and be optimized, because with the `push_back` to need to copy the structure in a pointer this can cause move, or a copy in some case.

With emplace_back you need to allocate the space of the struct one time, and this will summarize from:

push_back: 1 stack allocation + 1 copy to heap, and the copy cost O(N)
emplace_back: 1 allocation and this with reserve() cost O(1)

But yes, overall it is not a big deal.